packages/three-vrm-core/src/expressions/VRMExpressionMaterialColorBind.ts

```diff
@@ -75,84 +96,119 @@ export class VRMExpressionMaterialColorBind implements VRMExpressionBind {
      * The target color.
      */
     targetValue: THREE.Color;
+
+    /**
+     * The target alpha.
+     */
+    targetAlpha?: number;
   }) {
     this.material = material;
     this.type = type;
     this.targetValue = targetValue;
+    this.targetAlpha = targetAlpha ?? 1.0;
 
-    // init property name
-    const propertyNameMap = Object.entries(VRMExpressionMaterialColorBind._propertyNameMapMap).find(
-      ([distinguisher]) => {
-        return (material as any)[distinguisher] === true;
-      },
-    )?.[1];
-    const propertyName = propertyNameMap?.[type] ?? null;
+    // init bind state
+    const color = this._initColorBindState();
+    const alpha = this._initAlphaBindState();
+    this._state = { color, alpha };
+  }
 
-    if (propertyName == null) {
-      console.warn(
-        `Tried to add a material color bind to the material ${
-          material.name ?? '(no name)'
-        }, the type ${type} but the material or the type is not supported.`,
-      );
+  public applyWeight(weight: number): void {
+    const { color, alpha } = this._state;
 
-      this._state = null;
-    } else {
-      const target = (material as any)[propertyName] as THREE.Color;
+    if (color != null) {
+      const { propertyName, deltaValue } = color;
 
-      const initialValue = target.clone();
+      const target = (this.material as any)[propertyName] as THREE.Color;
```

Comment:
memo: wondering if change material  type to`THREE.MToonMaterial | THREE.MeshStandardMaterial | THREE.MeshBasicMaterial | THREE.Material` will improve types here